### PR TITLE
[#12048] Migrate AccountRequestSearchIndexingWorkerAction

### PR DESCRIPTION
**Pull request by domoberzin**
Part of #12048 

**Outline of Solution**

- Migrate `AccountRequestSearchIndexingWorkerAction.java` to handle both datastore and sql logic
- Created  `AccountRequestSearchIndexingWorkerActionIT.java`

### CODE REVIEW

File: src/main/java/teammates/ui/webapi/AccountRequestSearchIndexingWorkerAction.java [hunk 1]
```diff
@@ -5,25 +5,40 @@
...
+        AccountRequest accRequest = sqlLogic.getAccountRequest(email, institute);
+
+        //CHECKSTYLE.OFF:EmptyBlock
+        try {
+            sqlLogic.putAccountRequestDocument(accRequest);
+        } catch (SearchServiceException e) {
+            // Set an arbitrary retry code outside of the range 200-299 to trigger automatic retry
+            return new JsonResult("Failure", HttpStatus.SC_BAD_GATEWAY);
+        } catch (NullPointerException e) { }
+        //CHECKSTYLE.ON:EmptyBlock
```
Comment (cedricongjh) [inline src/main/java/teammates/ui/webapi/AccountRequestSearchIndexingWorkerAction.java:41]: apologises for not communicating this, but AccountRequest will not need to query both datastore and sql, hence this is unnecessary, and we simply need to change logic to sqlLogic here

**Review by cedricongjh — COMMENTED** [review]
one comment! thanks for the speedy PR

File: src/main/java/teammates/ui/webapi/AccountRequestSearchIndexingWorkerAction.java (hunk 1 above)
Comment (domoberzin) [reply src/main/java/teammates/ui/webapi/AccountRequestSearchIndexingWorkerAction.java:41]: noted on this, I assume that the tests in `AccountRequestSearchIndexingWorkerActionTest.java` can be disabled/deleted?

Comment (weiquu) [reply src/main/java/teammates/ui/webapi/AccountRequestSearchIndexingWorkerAction.java:41]: @domoberzin I think can remove, similar reasoning to discussion [here (+ comment above)](https://github.com/TEAMMATES/teammates/pull/12340#issuecomment-1501775407)

**Review by weiquu — APPROVED** [review]
LGTM!

**Review by cedricongjh — APPROVED** [review]
lgtm, thanks!

**Pull request merged**